src/lib/style-properties.html

```diff
@@ -331,6 +334,9 @@
           // parsedSelector fallback for 'shady' css build
           selectorToMatch = rule.transformedSelector || rule.parsedSelector;
         }
+        if (isRoot && hostScope === 'html') {
```

Comment:
`:root` for elements is supposed to match against the element itself, but the rule's transformedSelector will be something like `.x-foo:root`.

Should this be fixed in the selector transformer then? What about `:host > *`?